content/en/docs/gke/private-clusters.md

```diff
@@ -1,301 +1,235 @@
 
-Since private GKE can only access gcr.io, we need to mirror all images outside gcr.io for Kubeflow applications. We will use the `kfctl` tool to accomplish this.
+        ```
+        kpt cfg set ./upstream/manifests/gcp/v2/privateGKE/ log-firewalls true 
+        ```
 
+      * Change project to your project
 
-1. Set your user credentials. You only need to run this command once:
-   
-    ```
-    gcloud auth application-default login
-    ```
+      * Then look at the fields `jsonPayload.connection` this will tell you source and destination ips
+      * Based on the IPs try to figure out where the traffic is coming from (e.g. node to master) and
+        then match to appropriate firewall rules
 
-1. Inside your `${KFAPP}` directory create a local configuration file `mirror.yaml`  based on this [template](https://github.com/kubeflow/manifests/blob/master/experimental/mirror-images/gcp_template.yaml)
+      * For example
 
-    1. Change destination to your project gcr registry.
+         ```
+         connection: {
+          dest_ip: "172.16.0.34"    
+          dest_port: 443    
+          protocol: 6    
+          src_ip: "10.10.10.31"    
+          src_port: 60556    
+         }
+         disposition: "DENIED" 
+         ```
 
-1. Generate pipeline files to mirror images by running
-    
-    ```
-    cd ${KFAPP}
-    ./kfctl alpha mirror build mirror.yaml -V -o pipeline.yaml --gcb
-    ```
+         * The destination IP in this case is for a GKE master so the firewall rules are not configured to correctly allow
+           traffic to the master.
 
-    * If you want to use Tekton rather than Google Cloud Build(GCB) drop `--gcb` to emit a Tekton pipeline
-    * The instructions below assume you are using GCB
 
-1. Edit the couldbuild.yaml file
+* Common cause for networking related issue is is that some of the network resources (e.g. the Network, Routes, Firewall Rules, etc... ) don't get created
 
-    1. In the `images` section add
+     * This could be because a reference is incorrect (e.g. firewall rules reference the wrong network)
 
-         ```
-          - <registry domain>/<project_id>/docker.io/istio/proxy_init:1.1.6
-         ```
-     
-        * Replace `<registry domain>/<project_id>` with your registry      
+     * You can sanity check resources by doing kubectl describe and looking for errors.
 
-    1. Under `steps` section add
+       * [kubeflow/gcp-blueprints#38](https://github.com/kubeflow/gcp-blueprints/issues/38) is tracking
+          tools to automate this
 
-          ```
-            - args:
-            - build
-            - -t
-            - <registry domain>/<project id>/docker.io/istio/proxy_init:1.1.6
-            - --build-arg=INPUT_IMAGE=docker.io/istio/proxy_init:1.1.6
-            - .
-            name: gcr.io/cloud-builders/docker
-            waitFor:
-            - '-'  
-          ```
+* GCR images can't be pulled
```

Comment:
(Let's add a definition to "GCR", since it could be Google Cloud resources/Run/registry (depending on who's reading this))